### ISSUE TITLE
Problems with roberta-large on multi-classification task

### ISSUE THREAD

**Issue by igormis** [ThilinaRajapakse/simpletransformers#202]
**Describe the bug**
Problems when trying to train roberta-large on dataset with 3 labels

**To Reproduce**
```
model = ClassificationModel('roberta', 'roberta-large', num_labels=3, args=params) 
model.train_model(train_df, eval_df=eval_df)
```
where
```
params = {'no_cache': True, 'fp16': False, "max_seq_length": 60, 
          "gradient_accumulation_steps":1, "num_train_epochs": 5, 
          "learning_rate": 0.0001, 'overwrite_output_dir': True, 
          "evaluate_during_training": True, "evaluate_during_training_verbose": True, 
          "warmup_ratio": 0.1}
```

**Expected behavior**
The same code works with the bert-base model, but it has problems with the roberta-large, the running loss does not reduce

**Screenshots**
Epoch
40% 2/5 [23:33<34:42, 694.13s/it]
Current iteration
100% 1339/1339 [10:31<00:00, 2.20it/s]

Running loss: 0.740463Converting to features started. Cache is not used.

/usr/local/lib/python3.6/dist-packages/sklearn/metrics/_classification.py:900: RuntimeWarning: invalid value encountered in double_scalars
  mcc = cov_ytyp / np.sqrt(cov_ytyt * cov_ypyp)

{'mcc': 0.0, 'eval_loss': 0.5381082834888108}

Current iteration
100% 1339/1339 [11:43<00:00, 2.25it/s]

Running loss: 0.190201Converting to features started. Cache is not used.
{'mcc': 0.0, 'eval_loss': 0.5150014653110303}
Running loss: 0.801896Converting to features started. Cache is not used.
{'mcc': 0.0, 'eval_loss': 0.5175600507088352}

**Desktop (please complete the following information):**
 - Google Colab

**Additional context**
It seems that it takes into account only 2 labels..

**Comment by igormis** [comment]
Here are the model.args
`{'sliding_window': False, 'tie_value': 1, 'stride': 0.8, 'regression': False, 'output_dir': 'outputs/', 'best_model_dir': 'outputs/best_model', 'cache_dir': 'cache_dir/', 'fp16': False, 'fp16_opt_level': 'O1', 'max_seq_length': 60, 'train_batch_size': 8, 'gradient_accumulation_steps': 1, 'eval_batch_size': 8, 'num_train_epochs': 5, 'weight_decay': 0, 'learning_rate': 0.0001, 'adam_epsilon': 1e-08, 'warmup_ratio': 0.1, 'warmup_steps': 670, 'max_grad_norm': 1.0, 'do_lower_case': False, 'logging_steps': 50, 'save_steps': 2000, 'no_cache': True, 'save_model_every_epoch': True, 'evaluate_during_training': True, 'evaluate_during_training_steps': 2000, 'evaluate_during_training_verbose': True, 'use_cached_eval_features': True, 'save_eval_checkpoints': True, 'tensorboard_dir': None, 'overwrite_output_dir': True, 'reprocess_input_data': False, 'process_count': 1, 'n_gpu': 1, 'use_multiprocessing': True, 'silent': False, 'wandb_project': None, 'wandb_kwargs': {}, 'use_early_stopping': False, 'early_stopping_patience': 3, 'early_stopping_delta': 0, 'model_name': 'xlnet-large-cased', 'model_type': 'xlnet'}
`

**Comment by igormis** [comment]
It seems that everything is classified to the first class:
```
{'mcc': 0.0, 'eval_loss': 0.5218030937361818}
Test report               precision    recall  f1-score   support

           0       0.84      1.00      0.91      1590
           1       0.00      0.00      0.00       245
           2       0.00      0.00      0.00        56

    accuracy                           0.84      1891
   macro avg       0.28      0.33      0.30      1891
weighted avg       0.71      0.84      0.77      1891

[[1590    0    0]
 [ 245    0    0]
 [  56    0    0]]
```

**Comment by ThilinaRajapakse** [comment]
I think the issue may have something to do with the `max_seq_length` being 60. That's comparatively small.

I tested on the AG News dataset with 4 labels using roberta-large and got 0.93 mcc with the args given below.
```
train_args = {
    "learning_rate": 1e-5,
    "num_train_epochs": 2,
}
```

**Comment by igormis** [comment]
I tried with bert and it works, however I have problems with roberta. The reason why I am using `"max_seq_length": 60` is that the sentences are no longer than 60 tokens in my dataset (they are scrapped titles)...

**Comment by ThilinaRajapakse** [comment]
As this issue is not occurring for all datasets, I think it's unlikely to be an issue with the library. Can you try the tips mentioned in #199  and see whether they help? Early stopping and visualization might help to isolate the issue.

**Comment by yasminabelhadj** [comment]
Hello, I'm trying to train Roberta xlm on a dataset of african language, but the validation accuracy is always 0 and the loss doesn t improve. What may be the root of the problem? I tried 1.10-6 and 5.10-5 as learning rates and batch sizes of 1 , 4 and 8.
